diag_manager/fms_diag_reduction_methods.F90

```diff
@@ -0,0 +1,695 @@
+        call mpp_error( FATAL, "fms_diag_reduction_methods_mod::update_scalar_extremum type mismatch")
+      end select
+    type is (real(kind=r8_kind))
+      select type (buffer)
+      type is (real(kind=r8_kind))
+        if (flag .eq. 0) then
+        ! Update the buffer with the current minimum
+          where (mask(i-is+1+hi,j-js+1+hj,k,:) .AND. field_data(i-is+1+hi,j-js+1+hj,k,:) <&
+            buffer(i1,j1,k1,:,sample))
+            buffer(i1,j1,k1,:,sample) = field_data(i-is+1+hi,j-js+1+hj,k,:)
+          end where
+        else
+        ! Update the buffer with the current maximum
+          where (mask(i-is+1+hi,j-js+1+hj,k,:) .AND. field_data(i-is+1+hi,j-js+1+hj,k,:) >&
+            buffer(i1,j1,k1,:,sample))
+            buffer(i1,j1,k1,:,sample) = field_data(i-is+1+hi,j-js+1+hj,k,:)
+          end where
+        end if
+      class default
+        call mpp_error( FATAL, "fms_diag_reduction_methods_mod::update_scalar_extremum type mismatch")
+      end select
+    type is (integer(kind=i4_kind))
+      select type (buffer)
+      type is (integer(kind=i4_kind))
+        if (flag .eq. 0) then
+        ! Update the buffer with the current minimum
+          where (mask(i-is+1+hi,j-js+1+hj,k,:) .AND. field_data(i-is+1+hi,j-js+1+hj,k,:) <&
+            buffer(i1,j1,k1,:,sample))
+            buffer(i1,j1,k1,:,sample) = field_data(i-is+1+hi,j-js+1+hj,k,:)
+          end where
+        else
+        ! Update the buffer with the current maximum
+          where (mask(i-is+1+hi,j-js+1+hj,k,:) .AND. field_data(i-is+1+hi,j-js+1+hj,k,:) >&
+            buffer(i1,j1,k1,:,sample))
+            buffer(i1,j1,k1,:,sample) = field_data(i-is+1+hi,j-js+1+hj,k,:)
+          end where
+        end if
+      class default
+        call mpp_error( FATAL, "fms_diag_reduction_methods_mod::update_scalar_extremum type mismatch")
+      end select
+    type is (integer(kind=i8_kind))
+      select type (buffer)
+      type is (integer(kind=i8_kind))
+        if (flag .eq. 0) then
+        ! Update the buffer with the current minimum
+          where (mask(i-is+1+hi,j-js+1+hj,k,:) .AND. field_data(i-is+1+hi,j-js+1+hj,k,:) <&
+            buffer(i1,j1,k1,:,sample))
+            buffer(i1,j1,k1,:,sample) = field_data(i-is+1+hi,j-js+1+hj,k,:)
+          end where
+        else
+        ! Update the buffer with the current maximum
+          where (mask(i-is+1+hi,j-js+1+hj,k,:) .AND. field_data(i-is+1+hi,j-js+1+hj,k,:) >&
+            buffer(i1,j1,k1,:,sample))
+            buffer(i1,j1,k1,:,sample) = field_data(i-is+1+hi,j-js+1+hj,k,:)
+          end where
+        end if
+      class default
+        call mpp_error( FATAL, "fms_diag_reduction_methods_mod::update_scalar_extremum type mismatch")
+      end select
+    class default
+      call mpp_error( FATAL, "fms_diag_reduction_methods_mod::update_scalar_extremum unsupported field data type")
+    end select
+  end subroutine update_scalar_extremum
+
+  !> @brief Updates a chunk of buffer
```

Comment:
This routine and the one above it need more documentation.

Reply:
Added more information.